src/SourceLocator/Type/PhpInternalSourceLocator.php

```diff
@@ -41,8 +45,19 @@ private function getClassSource(Identifier $identifier): InternalLocatedSource|n
 
         /** @psalm-var class-string|trait-string $className */
         $className = $identifier->getName();
+        $aliasName = null;
+
+        if (class_exists($className, false)) {
+            $reflectionClass = new CoreReflectionClass($className);
+
+            if (strtolower($reflectionClass->getName()) !== strtolower($className)) {
+                $aliasName  = $className;
+                $className  = $reflectionClass->getName();
+                $identifier = new Identifier($className, $identifier->getType());
+            }
+        }
```

Comment:
Tricky stuff! So now we can handle aliases too :o

Reply:
Yes, there's probably first internal alias in PHP 8.4 :)